cmd/singularity/actions_test.go

```diff
@@ -319,6 +323,138 @@ func testRunFromURI(t *testing.T) {
 	}
 }
 
+// testPersistentOverlay test the --overlay function
+func testPersistentOverlay(t *testing.T) {
+	const squashfsImage = "squashfs.simg"
+	//  Create the overlay dir
+	cwd, err := os.Getwd()
+	if err != nil {
+		t.Fatal(err)
+	}
+	dir, err := ioutil.TempDir(cwd, "overlay_test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
+
+	// Create dirfs for squashfs
+	squashDir, err := ioutil.TempDir(cwd, "overlay_test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(squashDir)
+
+	content := []byte("temporary file's content")
+	tmpfile, err := ioutil.TempFile(squashDir, "bogus")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if _, err := tmpfile.Write(content); err != nil {
+		t.Fatal(err)
+	}
+	if err := tmpfile.Close(); err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(tmpfile.Name())
+
+	cmd := exec.Command("mksquashfs", squashDir, squashfsImage, "-noappend", "-all-root")
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	err = cmd.Run()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(squashfsImage)
+
+	//  Create the overlay ext3 fs
+	cmd = exec.Command("dd", "if=/dev/zero", "of=ext3_fs.img", "bs=1M", "count=768", "status=none")
+	cmd.Stdout = &out
+	err = cmd.Run()
+	if err != nil {
+		t.Fatal(err)
+	}
+	cmd = exec.Command("mkfs.ext3", "-q", "-F", "ext3_fs.img")
+	cmd.Stdout = &out
+	err = cmd.Run()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove("ext3_fs.img")
+
+	// create a file dir
+	t.Run("overlay_create", test.WithPrivilege(func(t *testing.T) {
+		_, stderr, exitCode, err := imageExec(t, "exec", opts{overlay: []string{dir}}, imagePath, []string{"touch", "/dir_overlay"})
```

Comment:
This should be a `mkdir` command

Reply:
is a file actually, but I want it under "/" that's why the `touch` cmd

Reply:
Then ... what does the "create a file dir" mean? It would just be a file ... nothing to do with a "directory"